statistics/cmsketch_util.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statistics
+
+import (
+	"time"
+
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/hack"
+)
+
+type datumMapCache struct {
+	datumMap map[hack.MutableString]types.Datum
+}
```

Comment:
Why do we need an extra `struct`?
I think it's the same as following:
```go
type datumMap map[hack.MutableString]types.Datum
```

Reply:
Personally, I think there's no need to add a new type just to wrap a simple `map` with `Get` and `Put`.
There are only two places using this, and it's not complicated.

Reply:
Actually, both definitions of ```datumMapCache``` are acceptable, but I just prefer the current form.